### ISSUE TITLE
cassandra cluster on K8s has a pod restarted and reaper is unable to get the new pod IP instead keep trying the old ip of the pod. 

### ISSUE THREAD

**Issue by sainathvadyala** [thelastpickle/cassandra-reaper#1105]
Cassandra cluster on K8s has a pod restarted and reaper is unable to get the new pod IP instead keep trying the old ip of the pod. This is memory storage. 

INFO   [2021-07-19 18:17:11,326] [pool-12-thread-5] i.c.j.JmxConnectionFactory - Adding new JMX Proxy for host 10.1.11:7199
ERROR  [2021-07-19 18:17:31,326] [pool-12-thread-5] i.c.j.JmxConnectionFactory - Failed creating a new JMX connection to 10.1.102.11:7199
java.lang.RuntimeException: io.cassandrareaper.ReaperException: Failure when establishing JMX connection to 10.1.102.11:7199

**Comment by sainathvadyala** [comment]
reaper version is 2.2.4

**Comment by rss04** [comment]
@sainathvadyala are you using IP instead of dns  for cassandra in reaper config ? if possible share your reaper config...deployment

**Comment by sainathvadyala** [comment]
No we are using DNS not the IP. Here is the deployment Yaml file.


apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  generation: 1
  labels:
    app: cassandra-reaper
  name: cassandra-reaper
  namespace: cassandra
spec:
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: cassandra-reaper
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: cassandra-reaper
    spec:
      containers:
      - env:
        - name: REAPER_JMX_AUTH_USERNAME
          value:
        - name: REAPER_JMX_AUTH_PASSWORD
          value: 
        - name: REAPER_STORAGE_TYPE
          value: memory
        - name: REAPER_CASS_CLUSTER_NAME
          value: production
        - name: REAPER_CASS_PORT
          value: "9042"
        - name: REAPER_SEGMENT_COUNT_PER_NODE
          value: "120"
        - name: REAPER_CASS_KEYSPACE
          value: reaper_db
        - name: REAPER_CASS_CONTACT_POINTS
          value: "[ cass-a.cassandra.svc.cluster.local ]"
        - name: REAPER_ENABLE_WEBUI_AUTH
          value: "true"
        - name: REAPER_CASS_AUTH_ENABLED
          value: "true"
        - name: REAPER_CASS_AUTH_USERNAME
          value: 
        - name: REAPER_CASS_AUTH_PASSWORD
          value:
        - name: REAPER_AUTO_SCHEDULING_ENABLED
          value: "true"
        image: thelastpickle/cassandra-reaper:2.2.4
        imagePullPolicy: Always
        name: cassandra-reaper
        ports:
        - containerPort: 8080
          name: http
          protocol: TCP
        - containerPort: 8081
          name: adminhttp
          protocol: TCP
        resources:
          requests:
            cpu: "1"
            memory: 1000Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      imagePullSecrets:
      - name: docker-reg-secret
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30

**Comment by rss04** [comment]
@sainathvadyala set your REAPER_STORAGE_TYPE=cassandra

The value below refers to the storage of reaper control data which in your case is cassandra. 
- name: REAPER_CASS_CONTACT_POINTS
value: "[ cass-a.cassandra.svc.cluster.local ]"
But your storage type is set to memory. 
- name: REAPER_STORAGE_TYPE
value: memory

**Comment by sainathvadyala** [comment]
Yes we purposely using memory storage and not using Cassandra storage.

**Comment by adejanovski** [comment]
@sainathvadyala, 
in order to get Reaper working against a kubernetes deployed Cassandra cluster, you'll need to provide a service as contact point (instead of a pod IP, as those get lost upon pod restart) and disable the seed list from being refreshed by setting the `REAPER_ENABLE_DYNAMIC_SEED_LIST` variable to `"false"` in Reaper's docker container.

**Comment by sainathvadyala** [comment]
Thanks @adejanovski for your reply. So I did make the change to the yaml with below and for the testing purpose I killed one of the pod and it recreated with a new IP. Now the existing repair which is already in progress have the old IP and cannot be changed to new IP and I still see the errors on reaper logs.

NFO   [2021-11-05 19:33:01,583] [production:122c63f0-3e69-11ec-80ef-c97c8d63302f] i.c.s.RepairRunner - Failed grabbing metrics from 10.111.0.19
java.util.concurrent.ExecutionException: io.cassandrareaper.ReaperException: no host could be reached through JMX

changes I did.
        - name: REAPER_ENABLE_DYNAMIC_SEED_LIST
          value: "false"
        - name: REAPER_CASS_CONTACT_POINTS
          value: '[ cassandra.cassandra.svc.cluster.local ]'  - removed pod name and just kept the service name.
